services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/application/ApplicationService.kt

```diff
@@ -45,6 +48,12 @@ open class ApplicationService(
             "Invalid geometry received when creating a new application for user $userId, reason = ${validationError.reason}, location = ${validationError.location}"
         }
 
+        val hanke =
+            hankeRepository.findByHankeTunnus(
+                application.hankeTunnus ?: throw HankeNotFoundException("")
```

Comment:
Could some other exception be more suitable here? In my opinion, trying to search with a null tunnus is more like a bad request than 404. Furthermore, the calling method (controller) already checks for this, so the error cannot actually happen here.

Reply:
This is the way other endpoints handle this by not giving information about existing hankkeet. Same point point as in the next one.

Reply:
This exception isn't throw when a hanke is not found, it's thrown when the caller has null in `application.hanketunnus`.

Reply:
Forced hankeTunnus to be non null.